### ISSUE TITLE
Chapter 9: Manual evaluation of gradients

### ISSUE THREAD

**Issue by SudhirGhandikota** [ageron/handson-ml#387]
Hello @ageron ,

I have been trying out tensor flow for the first time with the help of your great book but unfortunately I ran into an issue during chapter 9. I was trying to do the linear regression related examples (both theta computation using normal equations and Gradient Descent) but unfortunately the evaluation step just doesn't finish, it kind of gets stuck (with no errors). I tried using less number of data points (like 100) as well but it doesn't seem to work. My code is exactly the same the one in the book, also added below.

```python
housing = fetch_california_housing()
m, n = housing.data.shape
housing_bias = np.c_[np.ones((m,1)), housing.data]
X = tf.constant(housing_bias, dtype = tf.float32, name = "X")
y = tf.constant(housing.target.reshape(-1, 1), dtype = tf.float32, name = "y")
Xt = tf.transpose(X)
theta = tf.matmul(tf.matmul(tf.matrix_inverse(tf.matmul(Xt, X)),Xt),y)

with tf.Session() as sess:
    theta_value = theta.eval() # <== stuck
```

I am working on a Macbook Pro and I am aware that there will be no GPU support available but it still seems wrong that it doesn't work. Any help would be greatly appreciated!

Thanks & Regards,
Sudhir

**Comment by ageron** [comment]
Hi @SudhirGhandikota ,
Thanks for your question. Indeed, I see nothing wrong with this code. I can even run it without any issue by simply adding the right imports:

```python
from sklearn.datasets import fetch_california_housing
import tensorflow as tf
import numpy as np
```

At the end, I get this:

```pycon
>>> theta_value
array([[-3.6959320e+01],
       [ 4.3698898e-01],
       [ 9.4245886e-03],
       [-1.0791138e-01],
       [ 6.4842808e-01],
       [-3.9986235e-06],
       [-3.7866351e-03],
       [-4.2142656e-01],
       [-4.3467718e-01]], dtype=float32)
```

Everything looks good. So I'm not sure why your code gets stuck. Here are a few possible reasons I can think of:

* Installation problem: could you please check that you are using the latest version of TensorFlow? You can run `pip3 install -U tensorflow` in a terminal to upgrade to the latest version. Also upgrade NumPy and the other libraries.
* Different code: could you please copy/paste just the code above in a Python file like `tf_test.py` (make sure to include the imports, and perhaps add a `print(theta_value)` at the end), then run it by typing `python3 tf_test.py`.

If the problem persists, could you please try just starting a TensorFlow Session in a Python shell, like this:

```pycon
>>> import tensorflow as tf
>>> with tf.Session() as sess:
...     print(sess.run(tf.constant(1234)))
...
1234
```

Hope this helps,
Aurélien

**Comment by SudhirGhandikota** [comment]
Hello @ageron ,

Thanks a lot for the quick response!
Updating the tensorflow library using pip did the trick.
I am using an anaconda distribution and installed the package earlier today but it seems like the one available in anaconda was 1.12 while the latest one was 1.13.

Thanks again for your help!

Regards,
Sudhir
